File: modules/disk-info/src/DiskInfoModule.ts
import { NativeModule, requireNativeModule } from "expo";

export interface DiskInfoResult {
  total: number;
  available: number;
}

declare class DiskInfoModuleType extends NativeModule {
  getDiskInfo(): DiskInfoResult;
}

export default requireNativeModule<DiskInfoModuleType>("DiskInfo");

File: modules/disk-info/android/src/main/java/expo/modules/diskinfo/DiskInfoModule.kt
package expo.modules.diskinfo

import android.os.StatFs
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class DiskInfoModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("DiskInfo")

    Function("getDiskInfo") {
      val stat = StatFs(appContext.reactContext!!.filesDir.path)
      mapOf(
        "total" to stat.totalBytes,
        "available" to stat.availableBytes
      )
    }
  }
}
